src/folders.ts

```diff
@@ -79,10 +79,19 @@ export class CMakeToolsFolderController implements vscode.Disposable {
 
   /**
    * Get the CMakeTools instance associated with the given workspace folder, or undefined
-   * @param ws The workspace folder to search
+   * @param ws The workspace folder to search, or array of command and workspace path
    */
-  get(ws: vscode.WorkspaceFolder | undefined): CMakeToolsFolder | undefined {
-    return ws ? this._instances.get(ws.uri.fsPath) : ws;
+  get(ws: vscode.WorkspaceFolder | Array<string> | undefined): CMakeToolsFolder | undefined {
+    if (ws) {
+      if ((ws as vscode.WorkspaceFolder).uri) {
```

Comment:
Casting in TypeScript is just asserting that a type is what you say it is.  This won't do what you want. 

You need to use `instanceof` to check object types and `typeof` to check intrinsic types.  For example, we do this in cpptools:
```ts
export function isString(input: any): input is string {
    return typeof(input) === "string";
}

export function isArray(input: any): input is any[] {
    return input instanceof Array;
}

export function isArrayOfString(input: any): input is string[] {
    return isArray(input) && input.every(isString);
}
```

Reply:
Utilities extended, thank you!